lua/auto-dark-mode/health.lua

```diff
@@ -0,0 +1,102 @@
+local M = {}
+
+local uv = vim.uv or vim.loop
+
+local adm = require("auto-dark-mode")
+local interval = require("auto-dark-mode.interval")
+
+M.benchmark = function(iterations)
+	local result = {
+		avg = 0,
+		max = 0,
+		min = math.huge,
+		all_results = {},
+		stdout = nil,
+		stderr = nil,
+		parsed = nil,
+	}
+
+	for _ = 1, iterations do
+		local _start = uv.hrtime()
+		-- parsing the response is measured, but actually syncing the vim theme isn't performed
+		local function callback(_stdout, _stderr)
+			result.stdout = _stdout
+			result.stderr = _stderr
+			result.parsed = interval.parse_callback(_stdout, _stderr)
+			vim.schedule_wrap(function()
+				vim.print(result)
+			end)
+		end
+
+		local success, _ = pcall(interval.poll_dark_mode, callback)
+		-- bail early if polling fails
+		if not success then
+			break
+		end
+		local _end = uv.hrtime()
+
+		table.insert(result.all_results, (_end - _start) / 1e6)
+	end
+
+	local sum = 0
+	for _, v in pairs(result.all_results) do
+		result.max = result.max > v and result.max or v
+		result.min = result.min < v and result.min or v
+		sum = sum + v
+	end
+	result.avg = sum / iterations
+
+	return result
+end
+
+-- support for neovim < 0.9.0
```

Comment:
We can't actually test neovim 0.8.x with the busted test runner.
It looks like Ubuntu 24.04 and onward now have 0.9 [in their repositories](https://repology.org/project/neovim/versions), so  I'm not sure if supporting 0.8 is even worth it.

Reply:
I migrated the tests to use [Plenary's busted](https://github.com/nvim-lua/plenary.nvim/blob/master/TESTS_README.md) instead, things should work for earlier versions now.